### PR TITLE
r/aws_customer_gateway: Allow 4-byte ASNs

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/14027.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_customer_gateway: Support 4-byte ASN values for `bgp_asn` attribute
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCustomerGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSCustomerGateway_basic -timeout 120m
=== RUN   TestAccAWSCustomerGateway_basic
=== PAUSE TestAccAWSCustomerGateway_basic
=== CONT  TestAccAWSCustomerGateway_basic
--- PASS: TestAccAWSCustomerGateway_basic (62.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	62.588s
```

### CODE REVIEW

File: aws/resource_aws_customer_gateway.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func resourceAwsCustomerGateway() *schema.Resource {
 				Type:         schema.TypeInt,
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: validation.IntBetween(1, 65534),
+				ValidateFunc: validation.IntBetween(0, 4294967295),
```
Comment (bflad) [inline aws/resource_aws_customer_gateway.go:33]: Unfortunately this is an esoteric problem that isn't caught by CI (https://github.com/terraform-providers/terraform-provider-aws/issues/8988), but we'll need to do something different here to support 32-bit platforms:

```console
$ gox -os='linux darwin windows freebsd openbsd solaris' -arch='386 amd64 arm' -osarch='!darwin/arm !darwin/386' -ldflags '-s -w -X aws/version.ProviderVersion=99.99.99 -X aws/version.ProtocolVersion=4' -output 'results/{{.OS}}_{{.Arch}}/terraform-provider-aws_v99.99.99_x4' .
Number of parallel builds: 15

-->   solaris/amd64: github.com/terraform-providers/terraform-provider-aws
-->       linux/arm: github.com/terraform-providers/terraform-provider-aws
-->     freebsd/arm: github.com/terraform-providers/terraform-provider-aws
-->     freebsd/386: github.com/terraform-providers/terraform-provider-aws
-->     openbsd/386: github.com/terraform-providers/terraform-provider-aws
-->       linux/386: github.com/terraform-providers/terraform-provider-aws
-->     linux/amd64: github.com/terraform-providers/terraform-provider-aws
-->   windows/amd64: github.com/terraform-providers/terraform-provider-aws
-->    darwin/amd64: github.com/terraform-providers/terraform-provider-aws
-->   openbsd/amd64: github.com/terraform-providers/terraform-provider-aws
-->     windows/386: github.com/terraform-providers/terraform-provider-aws
-->   freebsd/amd64: github.com/terraform-providers/terraform-provider-aws

6 errors occurred:
--> openbsd/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> freebsd/arm error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> linux/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> linux/arm error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> windows/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> freebsd/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int
```

You can check it much quicker without `gox` via:

```console
$ GOOS=linux GOARCH=386 go build
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int
```

We can see how long the cross-compilation testing takes in GitHub Actions, although that was the original blocker for implementing it in TravisCI.

Comment (ewbankkit) [reply aws/resource_aws_customer_gateway.go:33]: Aah yes, this is similar to https://github.com/terraform-providers/terraform-provider-aws/pull/1888.
Is changing the attribute from `TypeInt` to `TypeString` a breaking change?

Comment (ewbankkit) [reply aws/resource_aws_customer_gateway.go:33]: In practice it seems not to be.

**Comment by ewbankkit** [comment]
```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCustomerGateway_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSCustomerGateway_ -timeout 120m
=== RUN   TestAccAWSCustomerGateway_basic
=== PAUSE TestAccAWSCustomerGateway_basic
=== RUN   TestAccAWSCustomerGateway_tags
=== PAUSE TestAccAWSCustomerGateway_tags
=== RUN   TestAccAWSCustomerGateway_similarAlreadyExists
=== PAUSE TestAccAWSCustomerGateway_similarAlreadyExists
=== RUN   TestAccAWSCustomerGateway_disappears
=== PAUSE TestAccAWSCustomerGateway_disappears
=== RUN   TestAccAWSCustomerGateway_4ByteAsn
=== PAUSE TestAccAWSCustomerGateway_4ByteAsn
=== CONT  TestAccAWSCustomerGateway_basic
=== CONT  TestAccAWSCustomerGateway_disappears
=== CONT  TestAccAWSCustomerGateway_similarAlreadyExists
=== CONT  TestAccAWSCustomerGateway_tags
=== CONT  TestAccAWSCustomerGateway_4ByteAsn
--- PASS: TestAccAWSCustomerGateway_disappears (29.92s)
--- PASS: TestAccAWSCustomerGateway_4ByteAsn (35.95s)
--- PASS: TestAccAWSCustomerGateway_similarAlreadyExists (43.40s)
--- PASS: TestAccAWSCustomerGateway_basic (63.07s)
--- PASS: TestAccAWSCustomerGateway_tags (66.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	66.472s
```

**Pull request merged**